src/common/redis_module/ray_redis_module.cc

```diff
@@ -716,24 +728,31 @@ int ReplyWithTask(RedisModuleCtx *ctx, RedisModuleString *task_id) {
     RedisModuleString *state = NULL;
     RedisModuleString *local_scheduler_id = NULL;
     RedisModuleString *task_spec = NULL;
-    RedisModule_HashGet(key, REDISMODULE_HASH_CFIELDS, "state", &state, "node",
-                        &local_scheduler_id, "TaskSpec", &task_spec, NULL);
+    RedisModule_HashGet(key, REDISMODULE_HASH_CFIELDS, "state", &state,
+                        "local_scheduler_id", &local_scheduler_id, "TaskSpec",
+                        &task_spec, NULL);
     if (state == NULL || local_scheduler_id == NULL || task_spec == NULL) {
       /* We must have either all fields or no fields. */
       RedisModule_CloseKey(key);
       return RedisModule_ReplyWithError(
           ctx, "Missing fields in the task table entry");
     }
 
-    int state_integer = ParseTaskState(state);
-    if (state_integer < 0) {
+    long long state_integer;
+    if (RedisModule_StringToLongLong(state, &state_integer) != REDISMODULE_OK) {
       RedisModule_CloseKey(key);
       RedisModule_FreeString(ctx, state);
       RedisModule_FreeString(ctx, local_scheduler_id);
       RedisModule_FreeString(ctx, task_spec);
       return RedisModule_ReplyWithError(ctx,
-                                        "Found invalid scheduling state (must "
-                                        "be an integer of width 2");
+                                        "Scheduling state must be integer.");
+    }
+    if (state_integer < 0) {
+      RedisModule_CloseKey(key);
+      RedisModule_FreeString(ctx, state);
+      RedisModule_FreeString(ctx, local_scheduler_id);
+      RedisModule_FreeString(ctx, task_spec);
+      return RedisModule_ReplyWithError(ctx, "Found invalid scheduling state.");
     }
 
     RedisModule_ReplyWithArray(ctx, 3);
```

Comment:
Can we also make the task reply a flatbuffer? It'd be great if we could use the same one for subscription messages and `task_table_get_task` replies.